File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.java
class Solution {
    public static int[] rearrangeArray(int[] arr) {
        int n=arr.length,k=0,j=0;
        int neg[]=new int[n/2];
        int pos[]=new int[n/2];
        for(int i=0;i<n;i++){
            if(arr[i]<0){
                neg[k++]=arr[i];
            }
            else pos[j++]=arr[i];

        }
        k=0;j=0;
        for(int i=0;i<n;i++){
            if(i%2!=0) arr[i]=neg[k++];
            else arr[i]=pos[j++];

        }
        return arr;
    }
}

File: 2149-rearrange-array-elements-by-sign/2149-rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int>pos;
        vector<int>neg;
        for(int i=0;i<nums.size();i++){
            if((nums[i]>0)==0){
                pos.push_back(nums[i]);
            }
            else{
                 neg.push_back(nums[i]);
            }
        }
        int j=0,k=0;
         for(int i=0;i<nums.size();i++){
            if((i&1)!=0){
                nums[i]=pos[j++];
            }
            else{
                nums[i]=neg[k++];
            }

         }
         return nums;

    }
};